libs/pilight/protocols/433.92/alecto_ws1700.c

```diff
@@ -75,13 +75,17 @@ static void parseCode(void) {
 		}
 	}
 
-	id = binToDecRev(binary, 0, 11);
+	header = binToDecRev(binary, 0, 3);
+	if (header != 5) {
+		return;
+	}
+	id = binToDecRev(binary, 4, 11);
 	battery = binary[12];
-	temperature = ((double)binToDecRev(binary, 18, 27));
+	temperature = (double)binToDecRev(binary, 16, 27);
 	humidity = (double)binToDecRev(binary, 28, 35);
 
-	if(temperature > 511) {
-		temperature -= 1023;
+	if (binary[16]) {
+	    temperature = temperature - 4096;
```

Comment:
Hi pilino1234,

Yes! you can. Normaly I would, just not paying attention.

As I’m quit new to the GitHub thingy: Can I just change the source and re-request for a "push-request”? Will the first one then disappear?

Please advice…!

Regards,

Willem Aandewiel
Almere

gsm: 0653 - 499 551
http: www.aandewiel.nl

---
:Core Wars: /n./  A game between `assembler' programs in a
   simulated machine, where the objective is to kill your opponent's
   program by overwriting it.  Popularized by A. K. Dewdney's column
   in "Scientific American" magazine, this was actually devised
   by Victor Vyssotsky, Robert Morris Sr., and Dennis Ritchie in the
   early 1960s (their original game was called `Darwin' and ran on a
   PDP-1 at Bell Labs).  See {core}.

> On 15 jun. 2016, at 13:33, pilino1234 <notifications@github.com> wrote:
> 
> In libs/pilight/protocols/433.92/alecto_ws1700.c <https://github.com/pilight/pilight/pull/287#discussion_r67145031>:
> 
> >  	humidity = (double)binToDecRev(binary, 28, 35);
> >  
> > -	if(temperature > 511) {
> > -		temperature -= 1023;
> > +	if (binary[16]) {
> > +	    temperature = temperature - 4096;
> You can use the -= operator here
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub <https://github.com/pilight/pilight/pull/287/files/f74c09ce59a0ffa3dfe12accbe3a1a1bf8580887#r67145031>, or mute the thread <https://github.com/notifications/unsubscribe/AFU6ExadIEE3sPTIcxA8Aut0mUpJUu8zks5qL-LzgaJpZM4I2DgO>.
>